### PR TITLE
Add in initial parser and transpiler code

### LINKED ISSUE #4: Store and take player decisions into account
Add the ability to branch the game's story based on previous decisions.

### LINKED ISSUE #10: Add a parser to easily create dialogues from plain text files
Right now the demo uses the JSON-like `.scene` files for the dialogue and scripting which would be cumbersome to write new scenes with. We should build a proper parser first before tackling issue #5.

An example script could be like this:
```
<<set_bg hallway>>
<<show Sophia left fade 1.5>>
// Comment here
Sophia, happy: "Hello!"
<<show Dani right slide 1>>
Dani, surprised: "Hi there!"
...
```
With a simple line-by-line loop, we can easily parse this into `.scene` files for the demo's dialogue system to read.

### PR DESCRIPTION
**Pull request by ameamenoame**
Add in the very messy code for the parser and the transpiler.

Related issue (if applicable): #10

### CODE REVIEW

File: godot/Parser/SceneTranspiler.gd [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+class_name SceneTranspiler
+extends Reference
+
+
+class DialogueTree:
+	# Store variables, jump points, etc.
+	const GLOBALS := {
+		jump_points = {},
+		variables = {}
+		}
```
Comment (NathanLovato) [inline godot/Parser/SceneTranspiler.gd]: If this is supposed to be available across the game, so shared across all scenes, you shouldn't define the constant here. It's never going to be a global either so I'd use two variables, say, on the main class, or on some singleton. But not on what's supposed to represent a single dialogue tree.

**Review by NathanLovato — CHANGES_REQUESTED** [review]
There's just one change I'd like with the GLOBALS property. You can remove it and store `variables` and `jump_points` in an autoload if that's convenient, otherwise in like `Main.gd` and pass it to `ScenePlayer` instances.

Best not present them as constants here as they're not constant (variables can be any value and change anytime with the player's decisions, jump points get updated with parsed scenes).

I'll change other things myself for the tutorials, like breaking some code into functions, but that's beyond the scope of your job. You did some great work already.

File: godot/Parser/SceneTranspiler.gd (hunk 1 above)
Comment (ameamenoame) [reply godot/Parser/SceneTranspiler.gd]: I've moved the `jump_points` somewhere else and used a singleton for handling variables.

**Comment by NathanLovato** [comment]
Great job! There are some things left to tweak and refactor but it's time to merge already.

**Pull request merged**